e2e/setup/e2e-setup.ts

```diff
@@ -0,0 +1,103 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import { config } from "dotenv-flow";
+
+config({
+  path: __dirname.concat("/../env/"),
+  // Disable warning messages in CI
+  silent: process.env.CI === "true",
+});
+
+const availableEnvironment = [
+  "ESS Dev-Next" as const,
+  "ESS Production" as const,
+  "ESS 1.1" as const,
+  "NSS" as const,
+];
+
+export type AvailableEnvironment = typeof availableEnvironment extends Array<
+  infer E
+>
+  ? E
+  : never;
+
+function isAvailableEnvironment(
+  environment: string | undefined
+): environment is AvailableEnvironment {
+  return availableEnvironment.includes(environment as AvailableEnvironment);
+}
+
+export interface TestingEnvironment {
+  clientId: string;
+  clientSecret: string;
+  environment: AvailableEnvironment;
+  // No environment-specific features at the moment
+  // feature: {}
+  idp: string;
+  pod: string;
+}
+
+function validateEnvVariables() {
+  let missing = false;
+  let missingMessage = "";
+
+  if (missing) {
+    throw new Error(`Environment variable missing: ${missingMessage}`);
+  }
+}
+
+export function getTestingEnvironment(): TestingEnvironment {
+  if (!isAvailableEnvironment(process.env.E2E_TEST_ENVIRONMENT)) {
+    throw new Error(`Unknown environment: ${process.env.E2E_TEST_ENVIRONMENT}`);
+  }
+  validateEnvVariables();
+  if (typeof process.env.E2E_TEST_POD !== "string") {
+    throw new Error("The environment variable E2E_TEST_POD is undefined.");
+  }
+  if (typeof process.env.E2E_TEST_IDP !== "string") {
+    throw new Error("The environment variable E2E_TEST_IDP is undefined.");
+  }
+  if (typeof process.env.E2E_TEST_CLIENT_ID !== "string") {
+    throw new Error(
+      "The environment variable E2E_TEST_CLIENT_ID is undefined."
+    );
+  }
+  if (typeof process.env.E2E_TEST_CLIENT_SECRET !== "string") {
+    throw new Error(
+      "The environment variable E2E_TEST_CLIENT_SECRET is undefined."
+    );
+  }
```

Comment:
Considering all this, I would probably extract the whole thing to an `asserts is TestingEnvironment` type guard.

Reply:
Done in 282b8eaa8b9c1087f8e482b96123cf808b6a0a6e